compiler/rustc_metadata/src/rmeta/encoder.rs

```diff
@@ -890,9 +890,11 @@ fn should_encode_mir(tcx: TyCtxt<'_>, def_id: LocalDefId) -> (bool, bool) {
             let needs_inline = (generics.requires_monomorphization(tcx)
                 || tcx.codegen_fn_attrs(def_id).requests_inline())
                 && tcx.sess.opts.output_types.should_codegen();
-            // The function has a `const` modifier or is annotated with `default_method_body_is_const`.
-            let is_const_fn = tcx.is_const_fn_raw(def_id.to_def_id())
-                || tcx.has_attr(def_id.to_def_id(), sym::default_method_body_is_const);
+            // The function has a `const` modifier or is in a `#[const_trait]`.
+            let mut is_const_fn = tcx.is_const_fn_raw(def_id.to_def_id());
+            if let Some(trait_) = tcx.trait_of_item(def_id.to_def_id()) {
+                is_const_fn = is_const_fn || tcx.has_attr(trait_, sym::const_trait);
```

Comment:
```suggestion
                is_const_fn ||= tcx.has_attr(trait_, sym::const_trait);
```

(I wish we had clippy enabled for this repository)